### DISCUSSION TITLE
印章检测与识别教程中label_list.txt文件以及KeyError问题

### DISCUSSION THREAD

**Discussion by dthin** [PaddlePaddle/PaddleOCR#13735] [Q&A]
按照这个教程一步步操作
https://paddlepaddle.github.io/PaddleOCR/applications/印章弯曲文字识别.html

![B$ ~0~WVD@$S~1AG49DGFS9](https://github.com/user-attachments/assets/9b999f2f-4a3f-4840-bbb4-16eadabbabcf)

按照教程里的python脚本处理数据集后生成的VOC格式目录，并不包含label_list.txt的文件
但是后续训练是需要这个文件的，教程里只做了简单说明，没说这个文件要怎么生成

尝试自己新建了文本并添加如下两个标签，放在对应位置
公章区域
其他

然后运行的时候会碰到KeyError的报错
Traceback (most recent call last):
  File "D:\paddle\PaddleDetection\tools\train.py", line 209, in <module>
    main()
  File "D:\paddle\PaddleDetection\tools\train.py", line 205, in main
    run(FLAGS, cfg)
  File "D:\paddle\PaddleDetection\tools\train.py", line 145, in run
    trainer = Trainer(cfg, mode='train')
  File "D:\paddle\PaddleDetection\ppdet\engine\trainer.py", line 100, in __init__
    self.loader = create('{}Reader'.format(capital_mode))(
  File "D:\paddle\PaddleDetection\ppdet\data\reader.py", line 168, in __call__
    self.dataset.parse_dataset()
  File "D:\paddle\PaddleDetection\ppdet\data\source\voc.py", line 161, in parse_dataset
    gt_class[i, 0] = cname2cid[cname]
KeyError: 'redseal'

有人可以帮忙讲解一下吗？这部分资料感觉比较少，找了很久找不到解答

**Comment by GreatV** [comment]
可以在aistudio上看看 https://aistudio.baidu.com/projectdetail/4983788

**Reply by dthin** [reply]
感谢回复 ，我今天回去试一下按这个教程走不走得通。

**Reply by dthin** [reply]
按着教程走，结果
configs\datasets目录下少一个seal_voc.yml文件，这个是怎么来的啊？是我少什么步骤了吗

**Reply by dthin** [reply]
那个教程我点运行，在aistudio上面， 很多流程已经走不通了。好几个脚本我看是直接运行的work目录下面的，然后work目录下面也没有

**Reply by GreatV** [reply]
可以看一下我们的开源大礼包里面提供的模型 https://paddlepaddle.github.io/PaddleOCR/ppocr/blog/ocr_book.html#_2

**Reply by GreatV** [reply]
看报错应该有个类是 redseal

**Reply by dthin** [reply]
redseal我大概知道了，是一开始教程里面的标注数据脚本，给印章标注的分类是readseal，我在label_list.txt里面添加这个单词程序就不会报错了。
这个学习大礼包我昨天有找飞浆小助手要过，我理解相当于是你们训练好的模型了，按照https://aistudio.baidu.com/projectdetail/4983788这个教程里面有个快速体验就是尝试操作了一下也有一点小问题。然后我现在碰到的困境是

1. 按照4983788这个教程使用大礼包里面的现成模型，并使用压缩包里面自带的ppyolo_mbv3_large_seal.yml配置文件，那么yml里面用到的configs\datasets\seal_voc.yml是缺失的，我看了下里面都是一些数据集的配置信息，不太清楚用大礼包模型的话，是不是还需要自己本地提供数据集，如果不需要的话，这个缺失的配置文件要怎么处理。

2. 上面那个readseal的字典问题我解决掉了，然后按自己数据集（类似教程里单张标注）跑了下训练，训练出来之后去infer原图发现都标注不上，不清楚是不是我的训练步骤或者数据有问题。

希望能帮忙解答下这两个方向的疑惑，任何一个都十分感谢，尤其是官方模型的，如果方法正确应该能比我自己训练先看到一点结果。
或者说有交流群吗？交流会方便一点，我官方搜到的交流群人满了，加不进去

**Reply by dthin** [reply]
还有就是PaddleDetection和PaddleOCR要怎么衔接啊，我看前者只是输出带标记的图片来展示，后者识别图章需要的应该是裁剪后的区域，如果需要中间程序做裁剪的话是不是PaddleDetection输出Json数据会更好一些

**Comment by dthin** [comment]
自己研究了下，做个回复，希望给碰到同样问题的人提供一点帮助。
按照4983788这个教程使用大礼包里面的现成模型，并使用压缩包里面自带的ppyolo_mbv3_large_seal.yml配置文件，那么yml里面用到的configs\datasets\seal_voc.yml是缺失的，这个seal_voc.yml的内容，其实就是官方印章教程里面教你训练时配置的一些训练集信息
metric: VOC
map_type: 11point
num_classes: 2

TrainDataset:
  !VOCDataSet
    dataset_dir: dataset/seal_VOC
    anno_path: train.txt
    label_list: label_list.txt
    data_fields: ['image', 'gt_bbox', 'gt_class', 'difficult']

EvalDataset:
  !VOCDataSet
    dataset_dir: dataset/seal_VOC
    anno_path: test.txt
    label_list: label_list.txt
    data_fields: ['image', 'gt_bbox', 'gt_class', 'difficult']

TestDataset:
  !ImageFolder
    anno_path: dataset/seal_VOC/label_list.txt

BBoxPostProcess:
  decode:
    name: YOLOBox
    conf_thresh: 0.005
    downsample_ratio: 32
    clip_bbox: true
    scale_x_y: 1.05
  nms:
    name: MultiClassNMS
    keep_top_k: 10  # 修改前100
    nms_threshold: 0.45
    nms_top_k: 100  # 修改前1000
    score_threshold: 0.005

可以自己补一个或者直接贴在ppyolo_mbv3_large_seal.yml里面把seal_voc.yml的文件引用删掉
如果只是想使用模型，不进行训练的话，TrainDataset和EvalDataset可以直接去掉，TestDataset和目录下的label_list.txt必须要有，而且官方模型里面对印章使用的关键字应该是seal，所以label_list.txt里面必须有seal这一行，没有的话会报keyError
